public/scripts.js

```diff
@@ -36,37 +36,10 @@ document.addEventListener('drop', (event) => {
     event.preventDefault();
     elements.wrapper.className = '';
 
-    // Read files
-    const files = event.dataTransfer.files;
-    if (files.length === 0) {
-        console.error('No files in drop event');
-        return;
-    }
-    for (const file of files) {
-        const reader = new FileReader();
-        reader.addEventListener('load', (event) => {
-            const fileName = file.name;
-            const fileType = file.type;
-            const fileData = event.target.result;
-
-            // JS still has no API to send multipart requests... m(
-            const boundary = random(32);
-            let data = '';
-            data += `--${boundary}\r\n`;
-            data += `content-disposition: form-data; name="file"; filename="${fileName}"\r\n`;
-            data += `content-type: ${fileType}\r\n`;
-            data += "\r\n";
-            data += fileData;
-            data += "\r\n";
-            data += `--${boundary}--`;
+    // Add files to form
+    const fileInput = document.querySelector('input[name="file"]');
+    fileInput.files = event.dataTransfer.files;
 
-            // Send request
-            const request = new XMLHttpRequest();
-            request.open('POST', '/', true);
-            request.setRequestHeader('Content-Type', `multipart/form-data; boundary=${boundary}`);
-            console.log(`Sending request for file ${fileName}`);
-            request.send(data);
-        });
-        reader.readAsBinaryString(file);
-    }
+    // Submit form
+    fileInput.closest('form').submit();
```

Comment:
Hmm can it be that this won't trigger `'submit'` event listeners? I can work around this by using `form.dispatchEvent(new Event('submit'));` in my follow up PR I guess.

Reply:
Apparently:

https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit

> This method is similar, but not identical to, activating a form's submit [<button>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button). When invoking this method directly, however: No [submit](https://developer.mozilla.org/en-US/docs/Web/API/HTMLFormElement/submit_event) event is raised. In particular, the form's onsubmit event handler is not run.

Another option to `dispatchEvent` seems to be [requestSubmit](https://html.spec.whatwg.org/multipage/forms.html#dom-form-requestsubmit-dev).

Reply:
Good otherwise?

Reply:
Yes.

Reply:
Also `requestSubmit` seems to work nicely.